Apps/WebClient/src/ClientApp/src/store/modules/vaccinationStatus/types.ts

```diff
@@ -34,11 +35,9 @@ export interface VaccinationStatusState {
         statusMessage: string;
     };
     authenticatedVaccineRecord: {
-        vaccinationRecord?: CovidVaccineRecord;
-        error?: ResultError;
-        status: LoadStatus;
-        statusMessage: string;
-        resultMessage: string;
+        activeHdid: string;
+        statusChanges: number;
```

Comment:
I think having these 2 properties could be code smells, but if it's working, I'm okay to merge now and address them later.

Relying on `activeHdid` is a little worrisome because the store is async.

`statusChanges` seems to be there just to have a reliable property that can be watched to see if `vaccinationRecords` has changed.  It should be possible to use `Vue.set` on `vaccinationRecords` in the mutations to make this redundant.

Reply:
Agreed.  I will re-examine when doing functional tests for AB#13782.

Reply:
I have done some more investigation. Unfortunately Vue does not support javascript Map. The use of Vue.set will not work. Looks like only javascript Object and Array have solutions for [reactivity.](https://v2.vuejs.org/v2/guide/reactivity.html#Change-Detection-Caveats), which is probably why we see the solution for preferences in user module.